src/components/HistoryTables/useAllMessages/index.ts

```diff
@@ -157,9 +157,21 @@ export const useAllMessages = (address: string, _offset: number = 0) => {
       !!allMessages?.messages
     ) {
       const confirmedCids = new Set(allMessages?.messages.map(msg => msg.cid))
-      return pendingMsgList
-        .filter(msg => !confirmedCids.has(msg.cid))
-        .sort((a, b) => Number(b.nonce) - Number(a.nonce))
+      return (
+        pendingMsgList
+          // Remove replaced messages
+          .filter((msg, i, arr) => {
+            return !arr.find(
+              m =>
+                m.nonce === msg.nonce &&
+                new BigNumber(m.gasPremium).isGreaterThan(msg.gasPremium)
```

Comment:
Two notes:

1. Let's move this logic into the `usePendingMessages` hook
2. We need to compare `from` and `nonce` - since an address sent _to_ this address could have the same nonce as once that was sent _from_ this address, and that's not a replaced transaction

Reply:
Good call about the `from` comparison!

Reply:
Isn't `usePendingMessages` auto-generated by graphql codegen? Can't browse easily since I'm not at my pc right now

Reply:
https://github.com/glifio/react-components/blob/primary/src/components/HistoryTables/useAllMessages/index.ts#L22 the pending messages logic is a bit more complex than that, you could actually implement this in `uniqueifyMessages` https://github.com/glifio/react-components/blob/primary/src/utils/uniqueifyMsgs/index.ts